web/components/logentry/site-map.jsx

```diff
@@ -0,0 +1,55 @@
+import geolib from 'geolib';
+import { GoogleMap, Marker, withGoogleMap } from 'react-google-maps';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class SiteMap extends React.Component {
+	render() {
+		// Paso de Cedral, Cozumel!
+		const defaultPosition = {lat: 20.354289, lng: -87.028272};
+
+		const gps = this.props.entry.gps || {};
+		let position, marker = null;
+
+		try {
+			if (gps.latitude && gps.longitude) {
+				position = {
+					lat: geolib.useDecimal(gps.latitude),
+					lng: geolib.useDecimal(gps.longitude)
+				};
+				marker =
+					<Marker
+						position={position}
+						key="dive site"
+						title={this.props.entry.site || 'dive site'}
+						defaultAnimation={2} />;
+			}
+		} catch(error) {
+			position = {};
+		}
+
+		if (!marker) {
+			position = defaultPosition;
+		}
+
+		return (
+			<GoogleMap
+				ref={this.props.onMapLoad}
+				defaultCenter={defaultPosition}
+				center={position}
+				defaultZoom={12}
+				onClick={this.props.onMapClick}>
+
+				{marker}
+
+			</GoogleMap>);
+	}
+}
+
+SiteMap.propTypes = {
+	onMapClick: PropTypes.func,
+	onMapLoad: PropTypes.func,
+	entry: PropTypes.object.isRequired
```

Comment:
We probably don't want this component updating on every update to the entry (API calls to Google and all.) Consider only passing in location, site, and GPS coords as props.